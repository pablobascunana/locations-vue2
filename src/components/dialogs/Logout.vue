<template>
  <v-dialog v-model="logoutDialogVisible" max-width="500px">
    <v-card>
      <v-card-title>
        {{ $t('logout.title') }}
        <v-spacer></v-spacer>
        <v-icon aria-label="Close" @click="closeDialog()">mdi-close</v-icon>
      </v-card-title>
      <v-card-text class="text-center mt-10">
        <p>{{ $t('logout.message') }}</p>
      </v-card-text>
      <v-card-actions>
        <v-spacer></v-spacer>
        <v-btn 
          id="logout" 
          color="primary"
          text
          @click="logout" 
        >{{ $t('generic.buttons.exit') }}
        </v-btn>
      </v-card-actions>
    </v-card>
  </v-dialog>
</template>

<script>
import { mapState } from 'vuex';
export default {
  name: 'logoutDialog',
  computed: {
    ...mapState('application', ['logoutDialogVisible'])
  },
  methods: {
    logout() {
      this.closeDialog();
      this.$store.commit('application/isLogin', false);
      this.$store.commit('resetStore');
      this.$router.push({ name: 'login' });
    },
    closeDialog() {
      this.$store.commit('application/isLogoutDialogVisible', false);
    }
  }
}
</script>
