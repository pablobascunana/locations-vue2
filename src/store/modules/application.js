function initialApplicationState() {
  return {
    isLogin: true,
    accessToken: undefined,
    refreshToken: undefined,
    addLocationDialogVisible: false,
    removeLocationDialogVisible: false
  }
}

const application = {
  namespaced: true,
  state: initialApplicationState,
  mutations: {
    resetApplicationStore(state) {
      const initState = initialApplicationState();
      Object.keys(initState).forEach(key => {
        state[key] = initState[key];
      })
    },
    isLogin(state, visible) {
      state.isLogin = visible;
    },
    setAccessToken(state, token) {
      state.accessToken = `Bearer ${token}`;
    },
    setRefreshToken(state, token) {
      state.refreshToken = token;
    },
    isAddLocationDialogVisible(state, visible) {
      state.addLocationDialogVisible = visible;
    },
    isRemoveLocationDialogVisible(state, visible) {
      state.removeLocationDialogVisible = visible
    }
  }
};

export default application;
  