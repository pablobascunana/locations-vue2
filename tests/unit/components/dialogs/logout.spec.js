import { createLocalVue, shallowMount } from '@vue/test-utils'

import VueRouter from 'vue-router'
import Vuetify from 'vuetify'
import Vuex from 'vuex'

import application from '@/store/modules/application.js'

import Logout from '@/components/dialogs/Logout.vue'

const localVue = createLocalVue();
localVue.use(VueRouter);
localVue.use(Vuex);
const router = new VueRouter();
let vuetify = new Vuetify();

describe('Logout', () => {
    let wrapper;
    let store;
    let state = {};
    let actions = {};
    let modules = {
        application
    };
    beforeEach(() => {
        store = new Vuex.Store({ state, actions, modules });

        wrapper = shallowMount(Logout, {
            localVue,
            vuetify,
            router,
            store,
            mocks: {
                $t: () => {'some specific text in any language'}
            }
        });
    });

    it("Should match snapshot", () => {
        expect(wrapper).toMatchSnapshot()    
    });

    it('Renders a vue instance', () => {
        expect(wrapper.isVueInstance()).toBe(true);
    });

    it('Call to logout function', () => {
        const logoutFn = jest.spyOn(wrapper.vm, 'logout');
        wrapper.vm.logout();
        expect(logoutFn).toBeCalled();
        expect(wrapper.vm.$store.state.application.isLogin).toBe(true);
        expect(wrapper.vm.$store.state.application.addLocationDialogVisible).toBe(false);
    });
});