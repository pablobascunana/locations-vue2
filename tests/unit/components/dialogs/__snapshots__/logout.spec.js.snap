// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout Should match snapshot 1`] = `
<v-dialog max-width="500px">
  <v-card>
    <v-card-title>

      <v-spacer></v-spacer>
      <v-icon id="close" aria-label="Close">mdi-close</v-icon>
    </v-card-title>
    <v-card-text class="text-center mt-10">
      <p></p>
    </v-card-text>
    <v-card-actions>
      <v-spacer></v-spacer>
      <v-btn id="dialogLogout" color="primary" text="">
      </v-btn>
    </v-card-actions>
  </v-card>
</v-dialog>
`;
